### ISSUE TITLE
using $ref sugar with $ref(number) conflicts with @typescript-eslint/restrict-plus-operands rule

### ISSUE THREAD

**Issue by leonzalion** [vuejs/eslint-plugin-vue#1839]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 8.12.0
- **eslint-plugin-vue version:** 8>5.0
- **Node version:** 16.14.1
- **Operating System:** macOS

**Please show your full configuration:**

```json5
module.exports = {
	extends: ['plugin:@typescript-eslint/recommended', 'plugin:vue/vue3-recommended'],
	plugins: ['@typescript-eslint'],
	parserOptions: {
		parser: '@typescript-eslint/parser',
		ecmaVersion: 2018,
		sourceType: 'module',
		extraFileExtensions: ['.vue'],
		project: './tsconfig.json'
	},
	rules: {
		'@typescript-eslint/restrict-plus-operands': 'error'
	}
}
```

**What did you do?**	

```vue
<script setup lang="ts">
const a = $ref(0);
console.log(a + 1);
</script>

<template>
  <div>div</div>
</template>
```

**What did you expect to happen?**
No lint error

**What actually happened?**

```
/Users/leonzalion/projects/eslint-bug/src/App.vue
  3:13  error  Operands of '+' operation must either be both strings or both numbers  @typescript-eslint/restrict-plus-operands

✖ 1 problem (1 error, 0 warnings)

 ELIFECYCLE  Command failed with exit code 1.
```
**Repository to reproduce this issue**

https://github.com/leonzalion/vue-ref-eslint-bug

**Comment by ota-meshi** [comment]
Thank you for posting issue.

The typescript-eslint/restrict-plus-operands rule doesn't seem to allow intersection type of number. (intersection type of string seem to be allowed.)

https://typescript-eslint.io/play/#ts=4.6.2&sourceType=module&code=MYewdgzgLgBBMF4YHJkwIbwBTQE4EswBzGAMhgG8BfASgChRIQAbAUwDpmQicYBqFHmQ0A3HQbhoMMIhgAWAEwZsYAK4BbAEatcZSrQlM2nblhkCAjKLpA&rules=N4IgAgLgngDgpgZwMYCcCWMIFpEBs0B2EA9CohOktjLgK4JYD28KAhgQCYIgBcIcKFIxQgAvkA&tsConfig=N4XyA

If you want to solve it, please suggest a feature to typescript-eslint.
I think the issue posted in the past will be helpful.
https://github.com/typescript-eslint/typescript-eslint/issues/2506

I close the issue, because we can't do anything with this repository.
